Adding hash for complete IPFS params

garethtdavies opened the pull request:
This closes #3865. I added the Sapling parameters to IPFS and are available via `/ipfs/QmeHWxaqS87cJFjJHXQAz4D82yuSTWPLxv9WfL4fQSu4cc/`. You might just want to add this to the existing directory and close this but I renamed to something more generic in addition to changing the hash.

You can see the output of `./zcutil/fetch-params.sh` as a result of this PR:

```
gareth@gareth-pc ~/g/zcash> ./zcutil/fetch-params.sh
Zcash - fetch-params.sh

This script will fetch the Zcash zkSNARK parameters and verify their
integrity with sha256sum.

If they already exist locally, it will exit now and do nothing else.

Retrieving (ipfs): /ipfs/QmeHWxaqS87cJFjJHXQAz4D82yuSTWPLxv9WfL4fQSu4cc/sprout-proving.key
Saving file(s) to /home/gareth/.zcash-params/sprout-proving.key.dl
868.01 MiB / 868.01 MiB [==================================================================================================================================================] 100.00% 11s
Download successful!
/home/gareth/.zcash-params/sprout-proving.key.dl: OK
renamed '/home/gareth/.zcash-params/sprout-proving.key.dl' -> '/home/gareth/.zcash-params/sprout-proving.key'

Retrieving (ipfs): /ipfs/QmeHWxaqS87cJFjJHXQAz4D82yuSTWPLxv9WfL4fQSu4cc/sprout-verifying.key
Saving file(s) to /home/gareth/.zcash-params/sprout-verifying.key.dl
1.42 KiB / 1.42 KiB [=======================================================================================================================================================] 100.00% 0s
Download successful!
/home/gareth/.zcash-params/sprout-verifying.key.dl: OK
renamed '/home/gareth/.zcash-params/sprout-verifying.key.dl' -> '/home/gareth/.zcash-params/sprout-verifying.key'

Retrieving (ipfs): /ipfs/QmeHWxaqS87cJFjJHXQAz4D82yuSTWPLxv9WfL4fQSu4cc/sapling-spend.params
Saving file(s) to /home/gareth/.zcash-params/sapling-spend.params.dl
45.74 MiB / 45.74 MiB [=====================================================================================================================================================] 100.00% 0s
Download successful!
/home/gareth/.zcash-params/sapling-spend.params.dl: OK
renamed '/home/gareth/.zcash-params/sapling-spend.params.dl' -> '/home/gareth/.zcash-params/sapling-spend.params'

Retrieving (ipfs): /ipfs/QmeHWxaqS87cJFjJHXQAz4D82yuSTWPLxv9WfL4fQSu4cc/sapling-output.params
Saving file(s) to /home/gareth/.zcash-params/sapling-output.params.dl
3.43 MiB / 3.43 MiB [=======================================================================================================================================================] 100.00% 0s
Download successful!
/home/gareth/.zcash-params/sapling-output.params.dl: OK
renamed '/home/gareth/.zcash-params/sapling-output.params.dl' -> '/home/gareth/.zcash-params/sapling-output.params'

Retrieving (ipfs): /ipfs/QmeHWxaqS87cJFjJHXQAz4D82yuSTWPLxv9WfL4fQSu4cc/sprout-groth16.params
Saving file(s) to /home/gareth/.zcash-params/sprout-groth16.params.dl
691.91 MiB / 691.91 MiB [===================================================================================================================================================] 100.00% 7s
Download successful!
/home/gareth/.zcash-params/sprout-groth16.params.dl: OK
renamed '/home/gareth/.zcash-params/sprout-groth16.params.dl' -> '/home/gareth/.zcash-params/sprout-groth16.params'
```

daira commented:
Note that this is a nontrivial conflict because #4098 changed each file to be chunked into two parts. The caching benefit of that does not apply to IPFS; it's up to you @garethtdavies whether you want to undo it for IPFS, or chunk the files in the same way.

garethtdavies commented:
While I appreciate it isn't necessary for IPFS, for simplicity, I simply chunked the files as above. Here's the output from running the script:

```
Zcash - fetch-params.sh

This script will fetch the Zcash zkSNARK parameters and verify their
integrity with sha256sum.

If they already exist locally, it will exit now and do nothing else.
The complete parameters are currently just under 1.7GB in size, so plan 
accordingly for your bandwidth constraints. If the Sprout parameters are
already present the additional Sapling parameters required are just under 
800MB in size. If the files are already present and have the correct 
sha256sum, no networking is used.

Creating params directory. For details about this directory, see:
/home/gareth/.zcash-params/README


Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sprout-proving.key.part.1
Saving file(s) to /home/gareth/.zcash-params/sprout-proving.key.dl.part.1
 500.00 MiB / 500.00 MiB [==================================================================================================================] 100.00% 1s
Download of part 1 successful!

Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sprout-proving.key.part.2
Saving file(s) to /home/gareth/.zcash-params/sprout-proving.key.dl.part.2
 368.01 MiB / 368.01 MiB [==================================================================================================================] 100.00% 0s
Download of part 2 successful!
/home/gareth/.zcash-params/sprout-proving.key.dl: OK
renamed '/home/gareth/.zcash-params/sprout-proving.key.dl' -> '/home/gareth/.zcash-params/sprout-proving.key'

Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sprout-verifying.key.part.1
Saving file(s) to /home/gareth/.zcash-params/sprout-verifying.key.dl.part.1
 1.42 KiB / 1.42 KiB [======================================================================================================================] 100.00% 0s
Download of part 1 successful!

Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sprout-verifying.key.part.2
Saving file(s) to /home/gareth/.zcash-params/sprout-verifying.key.dl.part.2
 0 B / ? [-----------------------------------------------------------------------------------------------------------------------------------------=] 0s
Download of part 2 successful!
/home/gareth/.zcash-params/sprout-verifying.key.dl: OK
renamed '/home/gareth/.zcash-params/sprout-verifying.key.dl' -> '/home/gareth/.zcash-params/sprout-verifying.key'

Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sapling-spend.params.part.1
Saving file(s) to /home/gareth/.zcash-params/sapling-spend.params.dl.part.1
 45.74 MiB / 45.74 MiB [====================================================================================================================] 100.00% 0s
Download of part 1 successful!

Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sapling-spend.params.part.2
Saving file(s) to /home/gareth/.zcash-params/sapling-spend.params.dl.part.2
 0 B / ? [-----------------------------------------------------------------------------------------------------------------------------------------=] 0s
Download of part 2 successful!
/home/gareth/.zcash-params/sapling-spend.params.dl: OK
renamed '/home/gareth/.zcash-params/sapling-spend.params.dl' -> '/home/gareth/.zcash-params/sapling-spend.params'

Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sapling-output.params.part.1
Saving file(s) to /home/gareth/.zcash-params/sapling-output.params.dl.part.1
 3.43 MiB / 3.43 MiB [======================================================================================================================] 100.00% 0s
Download of part 1 successful!

Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sapling-output.params.part.2
Saving file(s) to /home/gareth/.zcash-params/sapling-output.params.dl.part.2
 0 B / ? [-----------------------------------------------------------------------------------------------------------------------------------------=] 0s
Download of part 2 successful!
/home/gareth/.zcash-params/sapling-output.params.dl: OK
renamed '/home/gareth/.zcash-params/sapling-output.params.dl' -> '/home/gareth/.zcash-params/sapling-output.params'

Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sprout-groth16.params.part.1
Saving file(s) to /home/gareth/.zcash-params/sprout-groth16.params.dl.part.1
 500.00 MiB / 500.00 MiB [============================================================================================================] 100.00% 4h23m52s
Download of part 1 successful!

Retrieving (ipfs): /ipfs/QmXRHVGLQBiKwvNq7c2vPxAKz1zRVmMYbmt7G5TQss7tY7/sprout-groth16.params.part.2
Saving file(s) to /home/gareth/.zcash-params/sprout-groth16.params.dl.part.2
 191.91 MiB / 191.91 MiB [===============================================================================================================] 100.00% 1m29s
Download of part 2 successful!
/home/gareth/.zcash-params/sprout-groth16.params.dl: OK
renamed '/home/gareth/.zcash-params/sprout-groth16.params.dl' -> '/home/gareth/.zcash-params/sprout-groth16.params'
```

str4d commented:
Oh no, I clicked the green button! 😱 (Why was it green for me?) I'll make sure to keep an eye on the next PR being merged, to ensure this PR is also tested.
